src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,638 @@
+
+            if (Math.Abs(osmoregulationMultiplier.Value -
+                    WorldGenerationSettings.GetOsmoregulationMultiplier(preset)) > MathUtils.EPSILON)
+                continue;
+
+            if (freeGlucoseCloudButton.Pressed != WorldGenerationSettings.GetFreeGlucoseCloud(preset))
+                continue;
+
+            // If all values are equal to the values for a preset, use that preset
+            difficultyPresetButton.Selected = DifficultyPresetValueToIndex(preset);
+            difficultyPresetAdvancedButton.Selected = DifficultyPresetValueToIndex(preset);
+            return;
+        }
+
+        // If there is no preset with all values equal to the values set, use custom
+        difficultyPresetButton.Selected = DifficultyPresetValueToIndex(custom);
+        difficultyPresetAdvancedButton.Selected = DifficultyPresetValueToIndex(custom);
+    }
+
+    private void OnMPMultiplierValueChanged(double amount)
+    {
+        mpMultiplierReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.MPMultiplier = amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnAIMutationRateValueChanged(double amount)
+    {
+        aiMutationRateReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.AIMutationMultiplier = amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnCompoundDensityValueChanged(double amount)
+    {
+        compoundDensityReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.CompoundDensity = amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnPlayerDeathPopulationPenaltyValueChanged(double amount)
+    {
+        playerDeathPopulationPenaltyReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.PlayerDeathPopulationPenalty = amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnGlucoseDecayRateValueChanged(double percentage)
+    {
+        var percentageFormat = TranslationServer.Translate("PERCENTAGE_VALUE");
+        glucoseDecayRateReadout.Text = string.Format(CultureInfo.CurrentCulture, percentageFormat,
+            percentage);
+        settings.GlucoseDecay = percentage * 0.01;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnOsmoregulationMultiplierValueChanged(double amount)
+    {
+        osmoregulationMultiplierReadout.Text = amount.ToString(CultureInfo.CurrentCulture);
+        settings.OsmoregulationMultiplier = amount;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnFreeGlucoseCloudToggled(bool pressed)
+    {
+        settings.FreeGlucoseCloud = pressed;
+
+        UpdateDifficultyPreset();
+    }
+
+    private void OnLifeOriginSelected(int index)
+    {
+        // Set both buttons here as we only received a signal from one of them
+        lifeOriginButton.Selected = index;
+        lifeOriginButtonAdvanced.Selected = index;
+
+        settings.Origin = LifeOriginIndexToValue(index);
```

Comment:
Wouldn't it be far more efficient to cast these indexes directly to enum?
```csharp
settings.Origin = (WorldGenerationSettings.LifeOrigin)index;
```

Reply:
Did not know this was possible. Will definitely change that.